### PR TITLE
Feature/update toast

### PR DESCRIPTION
**Pull request by juo-dev**
Updated toast library

### CODE REVIEW

File: components/common/ButtonSale/index.tsx [hunk 1]
```diff
@@ -16,15 +16,15 @@ const ButtonSale = ({ order }: Props) => {
   const { account } = useContext(Web3Context)
   const { executeOrder, cancelOrder, loading, getOrders } = useContext(MarketplaceContext)
   const router = useRouter()
-  const { showError, showSuccess } = useToast()
+  const { showError, showLoading, toastUpdate } = useToast()
 
   async function onClickExecuteOrder() {
-    showSuccess('Execute order submitted')
+    const toastId = showLoading('Executing order, please wait')
```
[the remark this answers is not in the document]
Comment (karacurt) [reply components/common/ButtonSale/index.tsx:22]: suggestion to improve logs

File: utils/nfts.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const MouseHauntTokens = [
+    { addr: epicBooster.address, name: 'BMHTE' },
+    { addr: legendaryBooster.address, name: 'BMHTL' },
+    { addr: rareBooster.address, name: 'BMHTR' },
+    { addr: genesisBooster.address, name: 'BMHTG' },
+    { addr: mouseHero.address, name: 'MouseHero' }
+  ]
+
+  return MouseHauntTokens.find((f) => f.addr == address)?.name
+}
```
Comment (karacurt) [inline utils/nfts.ts:19]: There is `nftAddressToInfo: Record<string, NFT>` on `MarketplaceContext`, that do the same thing. You will just need to pass the address as key. Example: `nftAddressToInfo[nftAddress].name`.

File: utils/toast.ts [hunk 3]
```diff
@@ -1,22 +1,46 @@
...
+    toast.info(message, {
+      theme: 'colored'
+    })
+
+  const showLoading = (message: string): React.ReactText => toast.loading(message)
+
+  const toastUpdate = (toastId: React.ReactText, message: string) => {
+    toast.update(toastId, {
+      render: message,
+      type: 'success',
```
Comment (karacurt) [inline utils/toast.ts:30]: There will be a 'fail' update or only 'success' update?

**Review by karacurt — CHANGES_REQUESTED** [review]
Nice changes, just left a few comments.

File: utils/nfts.ts (hunk 2 above)
Comment (juo-dev) [reply utils/nfts.ts:19]: Right, I'll remove it

File: utils/toast.ts (hunk 3 above)
Comment (juo-dev) [reply utils/toast.ts:30]: Great!! I'll change it

**Review by karacurt — APPROVED** [review]
Nice! just a heads up to fix lint, prettier and build before merging

**Pull request merged**